addons/mrp/wizard/mrp_consumption_warning.py

```diff
@@ -40,6 +40,9 @@ def action_set_qty(self):
         for production in self.mrp_production_ids:
             for move in production.move_raw_ids:
                 rounding = move.product_uom.rounding
+                for line in self.mrp_consumption_warning_line_ids.filtered(lambda x: x.product_id == move.product_id):
+                    if float_compare(line.product_expected_qty_uom, move.product_uom_qty, precision_rounding=rounding) > 0:
+                        move.product_uom_qty = line.product_expected_qty_uom
```

Comment:
```suggestion
                        move.product_uom_qty = line.product_uom_id._compute_quantity(line.product_expected_qty_uom, move.product_uom)
```
Otherwise it would raise issues if the move is not in the same UoM as the product's default one (since quantities displayed on the wizard are all in that UoM)